### PR TITLE
Updating to use circe 0.9.0-M1

### PR DESCRIPTION
**Pull request by anusha-mallampati-ck**
Updating to use 0.9.0-M1 circe version and minor code fixes to be compatible with it.

### CODE REVIEW

File: shared/src/main/scala/io/circe/jackson/CirceJsonDeserializer.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ private[jackson] final class CirceJsonDeserializer(factory: TypeFactory, klass:
 
     val (maybeValue, nextContext) = (jp.getCurrentToken.id(): @switch) match {
       case JsonTokenId.ID_NUMBER_INT | JsonTokenId.ID_NUMBER_FLOAT =>
-        (Some(Json.JNumber(JsonBigDecimal(new BigDecimal(jp.getDecimalValue)))), parserContext)
```
Comment (travisbrown) [inline shared/src/main/scala/io/circe/jackson/CirceJsonDeserializer.scala]: I think we could simplify this to `JsonBigDecimal(jp.getDecimalValue)` now?

Comment (anusha-mallampati-ck) [inline shared/src/main/scala/io/circe/jackson/CirceJsonDeserializer.scala]: Oops. Simplified...thanks!

Comment (travisbrown) [inline shared/src/main/scala/io/circe/jackson/CirceJsonDeserializer.scala]: Excellent, thanks! I'll publish the update this evening (if not before).

**Comment by travisbrown** [comment]
Okay, 0.9.0-M1 release for circe-jackson is on its way to Maven Central, thanks again!

File: shared/src/main/scala/io/circe/jackson/CirceJsonDeserializer.scala (hunk 1 above)
Comment (anusha-mallampati-ck) [inline shared/src/main/scala/io/circe/jackson/CirceJsonDeserializer.scala]: Thank you!

**Comment by chris-vale-ck** [comment]
thanks @travisbrown

**Pull request merged**